java-client/src/main/java/org/opensearch/client/opensearch/generic/GenericBodies.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.opensearch.generic;
```

Comment:
It's really `json`, not `generic`, should we call it that? This way if we add other serializations in the future we have the opportunity to create a base class with a different name.

Reply:
> It's really `json`, not `generic`, should we call it that?

Not, it is not JSON to be fair, example is `Bulk` which is not JSON

Reply:
Bulk is "line-delimited JSON", so that's still JSON.

Reply:
> Bulk is "line-delimited JSON", so that's still JSON.

To elaborate on that, the content types that OpenSearch deals with include YAML, JSON, CBOR and SMILE , this is what server and client supports (`opensearch-java` only supports JSON but that is rather only client limitation)

Reply:
What if we subclasses the generics into YamlRequest, JsonRequest, etc.? For semantics and for passing the correct content-type.

Reply:
But why? That would basically add a tons of abstractions but not much value vs having factory methods to deal with request / response bodies.

```
GenericBodies  {
    GenericBody json(...)
    GenericBody yaml(...)
}

```

Reply:
I think I am stuck at wanting to write this:

```java
Response response = javaClient()
                .execute(
                    new Request(
                        "POST",
                        "/" + index + "/_search",
                        Collections.emptyList()
                    ).json(
                            "{"
                                + "    \"sort\": ["
                                + "        {"
                                + "            \"name\": \"asc\""
                                + "        }"
                                + "    ],"
                                + "    \"fields\": ["
                                + "        \"name\""
                                + "    ],"
                                + "    \"query\": {"
                                + "        \"bool\": {"
                                + "            \"filter\": ["
                                + "                {"
                                + "                    \"term\": {"
                                + "                        \"size\": \"huge\""
                                + "                    }"
                                + "                }"
                                + "            ]"
                                + "        }"
                                + "    },"
                                + "    \"_source\": true\n"
                                + "}"
                        )
                    )
                )
        ) 
```

The difference is that I removed the `Generic` references and moved `json` onto `Request`.

Reply:
Oh, that could be designed as a builder, sure! I was thinking about it but was focusing on getting the basics to work.

```
Requests
  .builder()
  .uri(...)
  .method(...)
  .json(...)
   ....
```
```

Reply:
Great. If the entire `GenericBlah` can be hidden, you're winning.